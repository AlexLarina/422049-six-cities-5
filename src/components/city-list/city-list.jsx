import React, {PureComponent} from "react";
import PropTypes from "prop-types";
import {connect} from "react-redux";
import {chooseCity} from "../../store/action.js";
import {CITIES} from "../../lib/const.js";

class CityList extends PureComponent {
  constructor(props) {
    super(props);
  }

  render() {
    const {city, handleClick} = this.props;

    return (
      <ul className="locations__list tabs__list">
        {CITIES.map((cityName, i) => {
          return (
            <li className="locations__item" key={`city-${i}`}>
              <a className={`locations__item-link tabs__item
                ${city === cityName ? `tabs__item--active` : ``}`}
              href="#"
              >
                <span onClick={handleClick}>{cityName}</span>
              </a>
            </li>
          );
        })}
      </ul>
    );
  }
}

const mapStateToProps = ({PROCESS}) => ({
  city: PROCESS.city,
});

const mapDispatchToProps = (dispatch) => ({
  handleClick(evt) {
    dispatch(chooseCity(evt.target.textContent));
    // @TO-DO заменить селектом по городам
    //dispatch(ActionCreator.getOfferList());
  }
});

CityList.propTypes = {
  city: PropTypes.string.isRequired,
  handleClick: PropTypes.func.isRequired
};

export {CityList};
export default connect(mapStateToProps, mapDispatchToProps)(CityList);
