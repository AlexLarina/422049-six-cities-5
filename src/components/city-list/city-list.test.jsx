import React from "react";
import renderer from "react-test-renderer";
import {CityList} from "./city-list.jsx";

it(`Render CityList`, () => {
  const tree = renderer
    .create(
        <CityList
          city={``}
          handleClick={() => {}}
        />
    )
    .toJSON();

  expect(tree).toMatchSnapshot();
});
