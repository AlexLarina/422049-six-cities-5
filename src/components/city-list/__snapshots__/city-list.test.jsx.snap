// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render CityList 1`] = `
<ul
  className="locations__list tabs__list"
>
  <li
    className="locations__item"
  >
    <a
      className="locations__item-link tabs__item
              "
      href="#"
    >
      <span
        onClick={[Function]}
      >
        Paris
      </span>
    </a>
  </li>
  <li
    className="locations__item"
  >
    <a
      className="locations__item-link tabs__item
              "
      href="#"
    >
      <span
        onClick={[Function]}
      >
        Cologne
      </span>
    </a>
  </li>
  <li
    className="locations__item"
  >
    <a
      className="locations__item-link tabs__item
              "
      href="#"
    >
      <span
        onClick={[Function]}
      >
        Brussels
      </span>
    </a>
  </li>
  <li
    className="locations__item"
  >
    <a
      className="locations__item-link tabs__item
              "
      href="#"
    >
      <span
        onClick={[Function]}
      >
        Amsterdam
      </span>
    </a>
  </li>
  <li
    className="locations__item"
  >
    <a
      className="locations__item-link tabs__item
              "
      href="#"
    >
      <span
        onClick={[Function]}
      >
        Hamburg
      </span>
    </a>
  </li>
  <li
    className="locations__item"
  >
    <a
      className="locations__item-link tabs__item
              "
      href="#"
    >
      <span
        onClick={[Function]}
      >
        Dusseldorf
      </span>
    </a>
  </li>
</ul>
`;
